### PR TITLE
Adds Express Best Security Practices

### PR DESCRIPTION
**Pull request by qwwqwwq**
- Use real session secret, secure session cookie.
- Use helmet middleware.
- Use global rate limiting per user and IP.
- Use CSRF protection for all angular and non-angular parts of the site.

### CODE REVIEW

**Comment by qwwqwwq** [comment]
ping

File: winterfell/middlewares/expressLimiter.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+ */
+function addLimiter(app) {
+  if (app.environment === Constants.environment.PROD) {
+    var limiter = expressLimiter(app, client);
+
+    limiter({
+      path: '/',
+      method: 'all',
+      lookup: ['connection.remoteAddress'],
+      // 360 request per hour per address for all pages
```
Comment (ahsu1230) [inline winterfell/middlewares/expressLimiter.js]: is 360/hr standard? I feel like it should be within a smaller window.
Attacks usually happen when a lot more requests are sent within a few minutes.

File: winterfell/middlewares/log.js [hunk 2]
```diff
@@ -42,6 +42,22 @@ module.exports.console = function(msg) {
   console.log(msg);
 };
 
+module.exports.info = function(msg) {
```
Comment (ahsu1230) [inline winterfell/middlewares/log.js]: nice I was thinking this too

Comment (ahsu1230) [inline winterfell/middlewares/log.js]: I know Android shortens it to
```
Log.i(...)
Log.d(...)
Log.w(...)
```
if we want to shorten things

Comment (qwwqwwq) [inline winterfell/middlewares/log.js]: My personal style is towards more verbose but easier to read for people new to the code, not sure what the node convention is though

**Pull request merged**